cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/ReadThroughBookCache.java

```diff
@@ -0,0 +1,135 @@
+        this.readAttrs = readAttrs;
+        this.schemaVersion = schemaVersion;
+    }
+
+    public BookInfo getBook (BookId bookId) throws CradleStorageException {
+        if (!books.containsKey(bookId)) {
+            logger.info("Book '{}' is absent in cache, trying to get it from DB", bookId.getName());
+            try {
+                books.put(bookId, loadBook(bookId));
+            } catch (CradleStorageException e) {
+                logger.warn("Could not load book named {}: {}", bookId.getName(), e);
+                throw e;
+            }
+        }
+
+        return books.get(bookId);
+    }
+
+    public Collection<PageInfo> loadPageInfo(BookId bookId) throws CradleStorageException
+    {
+        Collection<PageInfo> result = new ArrayList<>();
+        for (PageEntity pageEntity : ops.getOperators(bookId).getPageOperator().getAll(bookId.getName(), readAttrs))
+        {
+            if (pageEntity.getRemoved() == null)
+                result.add(pageEntity.toPageInfo());
+        }
+        return result;
+    }
+
+    public BookInfo loadBook (BookId bookId) throws CradleStorageException {
+        BookEntity bookEntity = ops.getCradleBookOperator().get(bookId.getName(), readAttrs);
+
+        if (bookEntity == null) {
+            throw new CradleStorageException(String.format("Book %s was not found in DB", bookId.getName()));
+        }
+
+        if (!bookEntity.getSchemaVersion().equals(schemaVersion)) {
+            throw new CradleStorageException(String.format(UNSUPPORTED_SCHEMA_VERSION_FORMAT,
+                    bookEntity.getName(),
+                    schemaVersion,
+                    bookEntity.getSchemaVersion()));
+        }
+
+        return processBookEntity(bookEntity);
+    }
+
+    private BookInfo processBookEntity (BookEntity entity) throws CradleStorageException {
+        BookId bookId = new BookId(entity.getName());
+        ops.addOperators(bookId, entity.getKeyspaceName());
+        Collection<PageInfo> pages = loadPageInfo(bookId);
+
+        return entity.toBookInfo(pages);
+    }
+
+    public Collection<BookInfo> loadBooks() throws CradleStorageException
+    {
+        Collection<BookInfo> result = new ArrayList<>();
+
+            for (BookEntity bookEntity : ops.getCradleBookOperator().getAll(readAttrs))
+            {
+                try {
+                    if(!bookEntity.getSchemaVersion().equals(schemaVersion)) {
+                        logger.warn(String.format(UNSUPPORTED_SCHEMA_VERSION_FORMAT,
+                                bookEntity.getName(),
+                                schemaVersion,
+                                bookEntity.getSchemaVersion()));
+                        continue;
+                    }
+
+                    result.add(processBookEntity(bookEntity));
+                } catch (CradleStorageException e) {
+                    logger.warn("Could not load book {}", bookEntity.getName());
+                }
+            }
```

Comment:
Indentation of this block is wrong. Too many spaces moved it to the right.

Reply:
I think we should not hold merging because of such kind of issues, considering this kind of formatting is more of the personal taste.

Reply:
Agreed. This won't hold merging anyway.